StartBonusActivation/handler.ts

```diff
@@ -289,21 +305,60 @@ const acquireLockForUserFamily = (
       },
       familyUID
     )
-  ).mapLeft(err =>
-    // consider any error a failure for lease already present
-    ResponseErrorConflict(
-      `Failed while acquiring lease for familiUID ${familyUID}: ${err.message}`
-    )
+  ).bimap(
+    err =>
+      // consider any error a failure for lease already prensent
+      ResponseErrorConflict(
+        `Failed while acquiring lease for familiUID ${familyUID}: ${err.message}`
+      ),
+    _ => familyUID
   );
 };
 
+/**
+ * Start a new instance of StartBonusActivationOrchestrator
+ *
+ * @param client an instance of durable function client
+ * @param bonusActivation a record of bonus activation
+ * @param fiscalCode the fiscal code of the requesting user. Needed to make a unique id for the orchestrator instance
+ *
+ * @returns either an internal error or the id of the created orchestrator
+ */
+const runStartBonusActivationOrchestrator = (
+  client: DurableOrchestrationClient,
+  bonusActivation: BonusActivationWithFamilyUID,
+  fiscalCode: FiscalCode
+): TaskEither<IResponseErrorInternal, string> =>
+  fromEither(OrchestratorInput.decode({ bonusActivation }))
+    .mapLeft(err =>
+      // validate input here, so we can make the http handler fail too. This shouldn't happen anyway
+      ResponseErrorInternal(
+        `Error validating orchestrator input: ${readableReport(err)}`
+      )
+    )
+    .chain(orchestratorInput =>
+      tryCatch(
+        () =>
+          client.startNew(
+            "StartBonusActivationOrchestrator",
+            makeStartBonusActivationOrchestratorId(fiscalCode),
+            orchestratorInput
+          ),
+        _ =>
+          // can it fail?
```

Comment:
yes it can and in case we must relase the lock before returning to the client (this states for every error that happens between locking and running the orchestrator)